src/lib/operators/table_scan.hpp

```diff
@@ -1,37 +1,47 @@
 #pragma once
 
+#include <unordered_map>
 #include "abstract_operator.hpp"
+#include "all_type_variant.hpp"
+#include "storage/chunk.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
 class TableScan : public AbstractOperator {
  public:
   TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id, const ScanType scan_type,
-            const AllTypeVariant search_value) {
-    // TODO(student) implement it in a source file and change this to a declaration.
-  }
+            const AllTypeVariant search_value);
 
-  ColumnID column_id() const {
-    // TODO(student) implement it in a source file and change this to a declaration.
-    Fail("Implementation missing.");
-  }
+  ColumnID column_id() const;
 
-  ScanType scan_type() const {
-    // TODO(student) implement it in a source file and change this to a declaration.
-    Fail("Implementation missing.");
-  }
+  ScanType scan_type() const;
 
-  const AllTypeVariant& search_value() const {
-    // TODO(student) implement it in a source file and change this to a declaration.
-    Fail("Implementation missing.");
-  }
+  const AllTypeVariant& search_value() const;
 
  protected:
-  std::shared_ptr<const Table> _on_execute() override {
-    // TODO(student) implement it in a source file and change this to a declaration.
-    Fail("Implementation missing.");
-  }
+  ColumnID _column_id;
```

Comment:
Again, comments to document/explain the functions would be nice :) 

![](https://media.giphy.com/media/6uGhT1O4sxpi8/giphy.gif)